airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplier.java

```diff
@@ -87,16 +84,70 @@ public JsonNode injectDestinationOAuthParameters(final UUID destinationDefinitio
   }
 
-  private ImmutableMap<String, Object> generateSourceMetadata(final UUID sourceDefinitionId)
-      throws JsonValidationException, ConfigNotFoundException, IOException {
-    final StandardSourceDefinition sourceDefinition = configRepository.getStandardSourceDefinition(sourceDefinitionId);
-    return TrackingMetadata.generateSourceDefinitionMetadata(sourceDefinition);
+  private static boolean injectOAuthParameters(final String connectorName,
+                                               final ConnectorSpecification spec,
+                                               final JsonNode oAuthParameters,
+                                               final JsonNode connectorConfig) {
+    if (hasOAuthConfigSpecification(spec)) {
+      if (!checkOAuthPredicate(spec.getAdvancedAuth().getPredicateKey(), spec.getAdvancedAuth().getPredicateValue(), connectorConfig)) {
+        return false;
+      }
+      // TODO: if we write a migration to flatten persisted configs in db, we don't need to flatten
+      // here see https://github.com/airbytehq/airbyte/issues/7624
+      final JsonNode flatOAuthParameters = MoreOAuthParameters.flattenOAuthConfig(oAuthParameters);
+      final JsonNode outputSpec = spec.getAdvancedAuth().getOauthConfigSpecification().getCompleteOauthServerOutputSpecification();
+      boolean result = false;
+      for (final String key : Jsons.keys(outputSpec)) {
+        final JsonNode node = outputSpec.get(key);
+        if (node.getNodeType() == OBJECT) {
+          final JsonNode pathNode = node.get(PATH_IN_CONNECTOR_CONFIG);
+          if (pathNode != null && pathNode.getNodeType() == ARRAY) {
+            final List<String> propertyPath = new ArrayList<>();
+            final ArrayNode arrayNode = (ArrayNode) pathNode;
+            for (int i = 0; i < arrayNode.size(); ++i) {
+              propertyPath.add(arrayNode.get(i).asText());
+            }
+            if (propertyPath.size() > 0) {
+              Jsons.replaceNestedValue(connectorConfig, propertyPath, flatOAuthParameters.get(key));
+              result = true;
+            } else {
+              LOGGER.error(String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification includes an invalid empty %s for %s",
+                  connectorName, PATH_IN_CONNECTOR_CONFIG, key));
+            }
+          } else {
+            LOGGER.error(
+                String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification does not declare an Array<String> %s for %s",
+                    connectorName, PATH_IN_CONNECTOR_CONFIG, key));
+          }
+        } else {
+          LOGGER.error(String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification does not declare an ObjectNode for %s",
+              connectorName, key));
+        }
+      }
+      return result;
+    } else {
+      MoreOAuthParameters.mergeJsons((ObjectNode) connectorConfig, (ObjectNode) oAuthParameters);
+      return true;
+    }
```

Comment:
Nitpick. There are quite a few `if` blocks here. They all have a large `if` branch, and a small `else` branch. This makes it harder to read the code. A cleaner approach is probably check for the `else` condition, and return early. So there will be fewer nested blocks.

Something like this:

```java
if (!hasOAuthConfigSpecification(spec)) {
  MoreOAuthParameters.mergeJsons((ObjectNode) connectorConfig, (ObjectNode) oAuthParameters);
  return true;
}

if (!checkOAuthPredicate(spec.getAdvancedAuth().getPredicateKey(), spec.getAdvancedAuth().getPredicateValue(), connectorConfig)) {
  return false;
}

// TODO: if we write a migration to flatten persisted configs in db, we don't need to flatten
// here see https://github.com/airbytehq/airbyte/issues/7624
final JsonNode flatOAuthParameters = MoreOAuthParameters.flattenOAuthConfig(oAuthParameters);
final JsonNode outputSpec = spec.getAdvancedAuth().getOauthConfigSpecification().getCompleteOauthServerOutputSpecification();
boolean result = false;
for (final String key : Jsons.keys(outputSpec)) {
  final JsonNode node = outputSpec.get(key);
  if (node.getNodeType() != OBJECT) {
    LOGGER.error(String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification does not declare an ObjectNode for %s",
        connectorName, key));
    continue;
  }

  final JsonNode pathNode = node.get(PATH_IN_CONNECTOR_CONFIG);
  if (pathNode == null || pathNode.getNodeType() != ARRAY) {
    LOGGER.error(
        String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification does not declare an Array<String> %s for %s",
            connectorName, PATH_IN_CONNECTOR_CONFIG, key));
    continue;
  }

  final List<String> propertyPath = new ArrayList<>();
  final ArrayNode arrayNode = (ArrayNode) pathNode;
  for (int i = 0; i < arrayNode.size(); ++i) {
    propertyPath.add(arrayNode.get(i).asText());
  }
  if (propertyPath.size() > 0) {
    Jsons.replaceNestedValue(connectorConfig, propertyPath, flatOAuthParameters.get(key));
    result = true;
  } else {
    LOGGER.error(String.format("In %s's advanced_auth spec, completeOAuthServerOutputSpecification includes an invalid empty %s for %s",
        connectorName, PATH_IN_CONNECTOR_CONFIG, key));
  }
}
return result;
```